sql/core/src/test/scala/org/apache/spark/sql/execution/SQLExecutionSuite.scala

```diff
@@ -119,6 +125,37 @@ class SQLExecutionSuite extends SparkFunSuite {
 
     spark.stop()
   }
+
+  test("SPARK-32813: Table scan should work in different thread") {
+    val executor1 = Executors.newSingleThreadExecutor()
+    val executor2 = Executors.newSingleThreadExecutor()
+    SparkSession.cleanupAnyExistingSession()
+
+    withTempDir { tempDir =>
```

Comment:
I would just use `withTemPath`

Reply:
SparkFunSuite has only withTempDir. I think it's not worth adding withTempPath here, so leave as is.